addons/web/static/src/js/core/widget.js

```diff
@@ -148,7 +148,7 @@ var Widget = core.Class.extend(mixins.PropertiesMixin, ServicesMixin, {
             }));
         }
         if (this.jsLibs || this.cssLibs || this.assetLibs) {
-            proms.push(ajax.loadLibs(this));
+            proms.push(ajax.loadLibs(this, this._rpc.bind(this)));
```

Comment:
what about adding a loadLibs function in the ajax service? That way, a 'load_libs' event would bubble up to the root widget, like for classical rpcs, and would could intercept this one as well to add your extra info. Moreover, it would be easily 'mockable' in tests.

Reply:
@aab-odoo It means I would have to change both the `loadLibs` of `ajax` to accept an extra context and the services mixin to add a new method.... and add my extension in website for that service call too. I can do it but I'll need a bit more convincing, I am not sure why that would be better, seems better to me that all `rpc` calls go through the `_rpc` method when they can IMHO? And that would already be mockable without having to do anything?

Reply:
I don't like changing the API of very low level functions (even if in this case it's backward compatible). Those functions should remain as simple as possible. Moreover, the rpcFunc you pass has a specific API (obviously), which differ from the (also basic, very low level) rpc method that is used by default.

Reply:
I will still have to change the API of the very low level function you mention, but ok I'll change.